CMakeLists.txt

```diff
@@ -98,10 +98,21 @@ function(RELATIVE_PROTOBUF_GENERATE_CPP SRCS HDRS ROOT_DIR)
     file(RELATIVE_PATH REL_DIR ${ROOT_DIR} ${FILE_DIR})
     set(OUTPUT_PROTO_DIR "${CMAKE_CURRENT_BINARY_DIR}/${REL_DIR}")
 
-    list(APPEND ${SRCS} "${OUTPUT_PROTO_DIR}/${FILE_WE}.pb.cc")
-    list(APPEND ${HDRS} "${OUTPUT_PROTO_DIR}/${FILE_WE}.pb.h")
+    if(ONNX_NAMESPACE STREQUAL "onnx")
+      set(OUTPUT_PB_HEADER "${OUTPUT_PROTO_DIR}/${FILE_WE}.pb.h")
+      set(OUTPUT_PB_SRC "${OUTPUT_PROTO_DIR}/${FILE_WE}.pb.cc")
+      set(GENERATED_PROTO "${OUTPUT_PROTO_DIR}/${FILE_WE}.proto")
+    else()
+      set(OUTPUT_PB_HEADER "${OUTPUT_PROTO_DIR}/${FILE_WE}_${ONNX_NAMESPACE}.pb.h")
+      set(OUTPUT_PB_SRC "${OUTPUT_PROTO_DIR}/${FILE_WE}_${ONNX_NAMESPACE}.pb.cc")
+      # We need this dummy header generated by gen_proto.py when ONNX_NAMESPACE
+      # is not onnx
+      list(APPEND ${HDRS} "${OUTPUT_PROTO_DIR}/${FILE_WE}.pb.h")
+      set(GENERATED_PROTO "${OUTPUT_PROTO_DIR}/${FILE_WE}_${ONNX_NAMESPACE}.proto")
```

Comment:
This is needed for both cases. files can be deleted or outdated.

Reply:
Both cases?